### DISCUSSION TITLE
Crash when starting in winPE

### DISCUSSION THREAD

**Discussion by npXXX** [AvaloniaUI/Avalonia#14040] [Q&A]
Hi,

I know Windows PE is not directly supported, but maybe there is an easy fix for my problem.
I have an application that just crashes on startup - while initializing Avalonia. See the exception below.

`Unhandled exception. System.Runtime.InteropServices.COMException (0x80040154): Class not registered (0x80040154 (REGDB_E_CLASSNOTREG))
at Avalonia.Win32.WinRT.NativeWinRTMethods.RoActivateInstance(IntPtr activatab1eC1assId)
at Avalonia.Win32.WinRT.NativeWinRTMethods.CreateInstance[T](String fullName)
at Avalonia.Win32.Win32PlatformSettings.GetColorValues()
at Avalonia.Application.RegisterServices()
at Avalonia.AppBuilder.SetupUnsafe()
at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, ShutdownMode shutdownMode)
at Netlnstall.Program.Main(String[] args)`

I was able to trace it down to what seems like missing COM support (more precise combase.dll seems to be missing/not registered), but the winPE packages for COM support are installed...
Maybe someone here had a similar problem and has a solution. Thanks in advance!

**Comment by maxkatz6** [comment]
It is possible to add try/catch or something here https://github.com/AvaloniaUI/Avalonia/blob/21ad94b98085767674204e7da71dd2b403759c38/src/Windows/Avalonia.Win32/Win32PlatformSettings.cs#L40
Contributions are welcome.

**Reply by mtniehaus** [reply]
I've been running an Avalonia app (.NET 8) on Windows PE recently and haven't had any crashing issues.  I am doing something a little odd though:  I am running a console app, and it later opens an Avalonia window.

This is the startup process I am using:

		static int Main(string[] args)
		{
			// Start the Avalonia UI window using a separate thread and wait for the window
			// to be created.
			_ = Task.Run(() =>
			{
				_ = BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
			});
			while (Global.ClientShellWindow == null)
			{
				System.Threading.Thread.Sleep(250);
			}

It spawns a separate thread to do the initialization, and waits for a window that is created by App.axaml to be created.

**Reply by mtniehaus** [reply]
One additional .NET 8 gotcha on Windows PE:

ICU, https://learn.microsoft.com/en-us/windows/win32/intl/international-components-for-unicode--icu-, is not present in Windows PE, so you have to tell .NET to use Nls instead (assuming somewhere in the code you used something locale related, which is pretty easy to do).  See https://learn.microsoft.com/en-us/dotnet/core/runtime-config/globalization#nls for how to do that (using System.Globalization.UseNls with RuntimeHostConfigurationOption in your .csproj file is easiest).

I also ran into another problem with Avalon.Vulkan.dll file missing, but that one was my own fault -- it looks like it wasn't required previously, but it definitely is now.
